### PR TITLE
[mono] Remove support for nint/nuint/nfloat from mini-native-types.c

### PR DESCRIPTION
**Pull request by akoeplinger**
The Xamarin.iOS types were replaced by the BCL versions.

### CODE REVIEW

**Comment by vargaz** [comment]
Are there any nugets etc. which still depend on the old types ?

**Comment by akoeplinger** [comment]
The .NET 6-based version of Xamarin.iOS isn't compatible with existing nugets so there's no concern there.

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -5742,23 +5487,7 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
...
 			int ret_mt = mint_type (m_class_get_byval_arg (klass));
-			if (mono_class_is_magic_int (klass) || mono_class_is_magic_float (klass)) {
-				g_assert (csignature->param_count == 1);
-#if SIZEOF_VOID_P == 8
-				if (mono_class_is_magic_int (klass) && td->sp [-1].type == STACK_TYPE_I4)
-					interp_add_conv (td, td->sp - 1, NULL, stack_type [ret_mt], MINT_CONV_I8_I4);
-				else if (mono_class_is_magic_float (klass) && td->sp [-1].type == STACK_TYPE_R4)
-					interp_add_conv (td, td->sp - 1, NULL, stack_type [ret_mt], MINT_CONV_R8_R4);
-#endif
-			} else if (klass == mono_defaults.int_class && csignature->param_count == 1)  {
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/transform.c:5753]: The removal of this seems unrelated to the rest of the change. Is it intentional ?

Comment (akoeplinger) [reply src/mono/mono/mini/interp/transform.c:5753]: @BrzVlad oops you're right, I meant to just kill the first if, not this one. Thanks!

**Comment by akoeplinger** [comment]
@vargaz would you mind taking another look? thanks

**Review by vargaz — APPROVED** [review]
